File: web-sdks/dart/lib/src/models/products/product_listing.dart
import 'part.dart';
import 'brand.dart';
import 'attribute_family.dart';

class ProductListing {
  final List<Part> parts;
  final List<Brand> brands;
  final List<String> shadows;
  final int partTypeId;
  final String partTypeName;
  final List<AttributeFamily> attributeFamilies;

  const ProductListing({
    required this.parts,
    required this.brands,
    required this.shadows,
    required this.partTypeId,
    required this.partTypeName,
    required this.attributeFamilies,
  });

  factory ProductListing.fromJson(Map<String, dynamic> json) => ProductListing(
    parts: (json['parts'] as List<dynamic>)
        .map((p) => Part.fromJson(p as Map<String, dynamic>))
        .toList(),
    brands: (json['brands'] as List<dynamic>)
        .map((b) => Brand.fromJson(b as Map<String, dynamic>))
        .toList(),
    shadows: (json['shadows'] as List<dynamic>)
        .map((s) => s as String)
        .toList(),
    partTypeId: json['part_type_id'],
    partTypeName: json['part_type_name'],
    attributeFamilies: (json['attribute_families'] as List<dynamic>)
        .map((af) => AttributeFamily.fromJson(af as Map<String, dynamic>))
        .toList(),
  );
}

File: web-sdks/javascript/src/models/products/product_listing.js
/**
 * Product listing model
 */

import { Part } from './part.js';

/**
 * Product listing - a collection of parts for a category
 */
export class ProductListing {
  constructor({ partTypeId, partTypeName, parts }) {
    this.partTypeId = partTypeId;
    this.partTypeName = partTypeName;
    this.parts = parts;
  }

  static fromJson(json) {
    return new ProductListing({
      partTypeId: json.part_type_id,
      partTypeName: json.part_type_name,
      parts: (json.parts || []).map(p => Part.fromJson(p)),
    });
  }

  toJson() {
    return {
      part_type_id: this.partTypeId,
      part_type_name: this.partTypeName,
      parts: this.parts,
    };
  }
}
